### ISSUE TITLE
`eth_sendRawTransaction` seems to not return the right contract address

### ISSUE THREAD

**Issue by gluneau** [AcalaNetwork/bodhi.js#133]
Behavior on ganache: Match!
![image](https://user-images.githubusercontent.com/2903505/142950388-4f616801-885b-4621-a415-a21bcfc9c273.png)

Behavior on bodhi: No Match!
![image](https://user-images.githubusercontent.com/2903505/142950499-0fac40c2-55d3-409c-8f72-7763e1bae3f5.png)

Ganache:
```
➜  hardhat npx hardhat run scripts/sample-script.js --network development
Greeter deployed to: 0x3d3593927228553b349767ABa68d4fb1514678CB
➜  hardhat 
➜  hardhat npx hardhat console --network development
Welcome to Node.js v14.18.1.
Type ".help" for more information.
> const Greeter = await ethers.getContractFactory("Greeter");
undefined
> const greeter = await Greeter.attach('0x3d3593927228553b349767ABa68d4fb1514678CB');
undefined
> await greeter.greet()
'Hello, Hardhat!'
> 
(To exit, press Ctrl+C again or Ctrl+D or type .exit)
> 
```

```
Listening on 127.0.0.1:8545
web3_clientVersion
eth_chainId
eth_blockNumber
eth_chainId
eth_chainId
eth_estimateGas
eth_getBlockByNumber
eth_getBlockByNumber
eth_gasPrice
eth_getTransactionCount
eth_chainId
eth_sendRawTransaction

  Transaction: 0xc9c911bfb1b4c9a6ff439a4c012d279dc696fa4120644056a968f8ee91b8a974
  Contract created: 0x3d3593927228553b349767aba68d4fb1514678cb
  Gas usage: 493826
  Block Number: 1
  Block Time: Mon Nov 22 2021 14:42:46 GMT-0500 (GMT-05:00)

eth_chainId
eth_getTransactionByHash
eth_chainId
eth_getTransactionReceipt
web3_clientVersion
eth_chainId
eth_chainId
eth_call
```

Bodhi:
```
➜  hardhat npx hardhat run scripts/sample-script.js --network development
Greeter deployed to: 0xD26e19913ca16B5B59aF7f07472f97cC9eA3f12B
➜  hardhat npx hardhat console --network development                     
Welcome to Node.js v14.18.1.
Type ".help" for more information.
> const Greeter = await ethers.getContractFactory("Greeter");
undefined
> const greeter = await Greeter.attach('0x3d3593927228553b349767aba68d4fb1514678cb');
undefined
> await greeter.greet()
'Hello, Hardhat!'
> 
```

```
[1637610589548] DEBUG (393790 on XPS-15-7590): incoming request
    jsonrpc: "2.0"
    method: "eth_getTransactionReceipt"
    params: [
      "0xfc78254710528e5512b56cb65fd82b20e07096c8613c1818f83b756c555658da"
    ]
    id: 24
[1637610589559] DEBUG (393790 on XPS-15-7590): request completed
    id: 24
    jsonrpc: "2.0"
    result: {
      "to": null,
      "from": "0x75e480db528101a381ce68544611c169ad7eb342",
      "contractAddress": "0x3d3593927228553b349767aba68d4fb1514678cb",
      "transactionIndex": "2",
      "gasUsed": "0",
      "logsBloom": "0x",
      "blockHash": "0x933f845f22fda331412c2d0447e0b5650a9c2bacfbf8f342b23fb78ffd36811e",
      "transactionHash": "0xfc78254710528e5512b56cb65fd82b20e07096c8613c1818f83b756c555658da",
      "logs": [],
      "blockNumber": "37",
      "cumulativeGasUsed": "0",
      "type": "0",
      "status": "1",
      "effectiveGasPrice": "0x1",
      "confirmations": "0x1"
    }
```

**Comment by shunjizhan** [comment]
yup I also notice this, investigating...

**Comment by shunjizhan** [comment]
fixed by #136
